plugins/soundsourcem4a/audiosourcem4a.cpp

```diff
@@ -0,0 +1,337 @@
+}
+
+AudioSource::diff_type AudioSourceM4A::seekSampleFrame(diff_type frameIndex) {
+    if (m_curFrameIndex != frameIndex) {
+        const MP4SampleId sampleBlockId =
+            kMinSampleBlockId + (frameIndex / kFramesPerSampleBlock);
+        if (!isValidSampleBlockId(sampleBlockId)) {
+            return m_curFrameIndex;
+        }
+        if ((frameIndex < m_curFrameIndex) || // seeking backwards?
+            !isValidSampleBlockId(m_curSampleBlockId) || // invalid seek position?
+            (sampleBlockId > (m_curSampleBlockId + kNumberOfPrefetchSampleBlocks))) { // jumping forward?
+            // Restart decoding one or more blocks of samples backwards
+            // from the calculated starting block to avoid audible glitches.
+            // Implementation note: The type MP4SampleId is unsigned so we
+            // need to be careful when subtracting!
+            if ((kMinSampleBlockId + kNumberOfPrefetchSampleBlocks) < sampleBlockId) {
+                m_curSampleBlockId = sampleBlockId - kNumberOfPrefetchSampleBlocks;
+            } else {
+                m_curSampleBlockId = kMinSampleBlockId;
+            }
+            NeAACDecPostSeekReset(m_hDecoder, m_curSampleBlockId);
+            m_curFrameIndex = (m_curSampleBlockId - kMinSampleBlockId) * kFramesPerSampleBlock;
+            // discard input buffer
+            m_inputBufferOffset = 0;
+            m_inputBufferLength = 0;
+        }
+        // decoding starts before the actual target position
+        DEBUG_ASSERT(m_curFrameIndex <= frameIndex);
+        const size_type prefetchFrameCount = frameIndex - m_curFrameIndex;
+        // prefetch (decode and discard) all samples up to the target position
+        DEBUG_ASSERT(frames2samples(prefetchFrameCount) <= m_prefetchSampleBuffer.size());
+        readSampleFrames(prefetchFrameCount, &m_prefetchSampleBuffer[0]);
+    }
+    DEBUG_ASSERT(m_curFrameIndex == frameIndex);
+    return frameIndex;
+}
+
+AudioSource::size_type AudioSourceM4A::readSampleFrames(
+        size_type numberOfFrames, sample_type* sampleBuffer) {
+    if (!isValidSampleBlockId(m_curSampleBlockId)) {
+        qWarning() << "Invalid MP4 sample block" << m_curSampleBlockId;
+        return 0;
+    }
+    sample_type* pSampleBuffer = sampleBuffer;
+    size_type numberOfFramesRemaining = numberOfFrames;
+    while (0 < numberOfFramesRemaining) {
+        DEBUG_ASSERT(m_inputBufferOffset <= m_inputBufferLength);
+        if (m_inputBufferOffset >= m_inputBufferLength) {
+            // reset input buffer
+            m_inputBufferOffset = 0;
+            m_inputBufferLength = 0;
+            if (isValidSampleBlockId(m_curSampleBlockId)) {
+                // fill input buffer with next block of samples
+                u_int8_t* pInputBuffer = &m_inputBuffer[0];
+                u_int32_t inputBufferLength = m_inputBuffer.size(); // in/out parameter
+                if (!MP4ReadSample(
+                    m_hFile, m_trackId, m_curSampleBlockId,
```

Comment:
Is this mentioned in the coding guidelines? I could not find it anywhere, there is only a 4 spaces indentation rule: http://mixxx.org/wiki/doku.php/coding_guidelines

So I just changed my auto-formatter from 8 to 4 and re-formatted a lot of files manually :(

Please, provide some mechanism for auto-formatting, because 'm really getting tired of reformatting my code once and again. Otherwise I wil spend my precious time elsewhere ;) Sorry, Daniel, no personal offense, just a hint for the Mixxx team not to discourage motivated contributors.